File: test/org/cerion/symcalc/expression/function/list/TallyTest.java
package org.cerion.symcalc.expression.function.list;

import org.cerion.symcalc.expression.Expr;
import org.cerion.symcalc.expression.ListExpr;
import org.cerion.symcalc.expression.VarExpr;
import org.cerion.symcalc.expression.number.IntegerNum;
import org.junit.Test;

import static org.junit.Assert.*;

public class TallyTest {

    @Test
    public void basic() {
        assertEquals(new ListExpr(new ListExpr(new VarExpr("a"), new IntegerNum(3))),
                new Tally(new ListExpr(new VarExpr("a"), new VarExpr("a"), new VarExpr("a"))).eval());

        Expr expected = new ListExpr(new ListExpr(new IntegerNum(5), new IntegerNum(2)), new ListExpr(new VarExpr("a"), new IntegerNum(1)), new ListExpr(new IntegerNum(2), new IntegerNum(1)));
        Expr actual = new Tally(new ListExpr(new IntegerNum(5), new VarExpr("a"), new IntegerNum(5), new IntegerNum(2))).eval();
        assertEquals(expected, actual);
    }
}

File: test/expression/function/list/TallyTest.kt
package org.cerion.symcalc.expression.function.list

import org.cerion.symcalc.expression.ListExpr
import org.cerion.symcalc.expression.VarExpr
import org.cerion.symcalc.expression.number.IntegerNum
import org.junit.Assert.assertEquals
import org.junit.Test

class TallyTest {

    @Test
    fun basic() {
        assertEquals(ListExpr(ListExpr(VarExpr("a"), IntegerNum(3))),
                Tally(ListExpr(VarExpr("a"), VarExpr("a"), VarExpr("a"))).eval())

        val expected = ListExpr(ListExpr(IntegerNum(5), IntegerNum(2)), ListExpr(VarExpr("a"), IntegerNum(1)), ListExpr(IntegerNum(2), IntegerNum(1)))
        val actual = Tally(ListExpr(IntegerNum(5), VarExpr("a"), IntegerNum(5), IntegerNum(2))).eval()
        assertEquals(expected, actual)
    }
}